batch/batch/front_end/validate.py

```diff
@@ -41,86 +41,76 @@
-                },
-                'jvm': {
-                    required('jar_spec'): keyed(
-                        {required('type'): oneof('git_revision', 'jar_url'), required('value'): str_type}
-                    ),
-                    required('command'): listof(str_type),
-                    'profile': bool_type,
-                },
+job_validator = keyed({
+    'always_copy_output': bool_type,
+    'always_run': bool_type,
+    'attributes': dictof(str_type),
+    'env': listof(keyed({'name': str_type, 'value': str_type})),
+    'cloudfuse': listof(
+        keyed({
+            required('bucket'): non_empty_str_type,
+            required('mount_path'): non_empty_str_type,
+            required('read_only'): bool_type,
+        })
+    ),
+    'input_files': listof(keyed({required('from'): str_type, required('to'): str_type})),
+    required('job_id'): int_type,
+    'mount_tokens': bool_type,
+    'network': oneof('public', 'private'),
+    'unconfined': bool_type,
+    'output_files': listof(keyed({required('from'): str_type, required('to'): str_type})),
+    'parent_ids': listof(int_type),
+    'absolute_parent_ids': listof(int_type),
+    'in_update_parent_ids': listof(int_type),
+    'port': int_type,
+    required('process'): switch(
+        'type',
+        {
+            'docker': {
+                required('command'): listof(str_type),
+                required('image'): image_str,
+                'mount_docker_socket': bool_type,  # DEPRECATED
             },
-        ),
-        'regions': listof(str_type),
-        'requester_pays_project': str_type,
-        'resources': keyed(
-            {
-                'memory': anyof(regex(MEMORY_REGEXPAT, MEMORY_REGEX), oneof(*memory_types)),
-                'cpu': regex(CPU_REGEXPAT, CPU_REGEX),
-                'storage': regex(STORAGE_REGEXPAT, STORAGE_REGEX),
-                'machine_type': str_type,
-                'pool_label': str_type,
-                'preemptible': bool_type,
-            }
-        ),
-        'secrets': listof(
-            keyed({required('namespace'): k8s_str, required('name'): k8s_str, required('mount_path'): str_type})
-        ),
-        'service_account': keyed({required('namespace'): k8s_str, required('name'): k8s_str}),
-        'timeout': numeric(**{"x > 0": lambda x: x > 0}),
-        'user_code': str_type,
-    }
-)
-
-batch_validator = keyed(
-    {
-        'attributes': nullable(dictof(str_type)),
-        required('billing_project'): str_type,
-        'callback': nullable(str_type),
-        required('n_jobs'): int_type,
-        required('token'): str_type,
-        'cancel_after_n_failures': nullable(numeric(**{"x > 0": lambda x: isinstance(x, int) and x > 0})),
-    }
-)
-
-batch_update_validator = keyed(
-    {
-        required('token'): str_type,
-        required('n_jobs'): numeric(**{"x > 0": lambda x: isinstance(x, int) and x > 0}),
-    }
-)
+            'jvm': {
```

Comment:
Applied incoming
	LN 74 new 'jvm'
	'pool_label': str_type, removed

was:
```
<<<<<<< HEAD
        ),
        'regions': listof(str_type),
        'requester_pays_project': str_type,
        'resources': keyed(
            {
                'memory': anyof(regex(MEMORY_REGEXPAT, MEMORY_REGEX), oneof(*memory_types)),
                'cpu': regex(CPU_REGEXPAT, CPU_REGEX),
                'storage': regex(STORAGE_REGEXPAT, STORAGE_REGEX),
                'machine_type': str_type,
                'pool_label': str_type,
                'preemptible': bool_type,
            }
        ),
        'secrets': listof(
            keyed({required('namespace'): k8s_str, required('name'): k8s_str, required('mount_path'): str_type})
        ),
        'service_account': keyed({required('namespace'): k8s_str, required('name'): k8s_str}),
        'timeout': numeric(**{"x > 0": lambda x: x > 0}),
        'user_code': str_type,
    }
)
=======
```

Reply:
I looked at this one again, and noticed that `pool_label` appears in the `HEAD` part but not the merge-base part. I then remembered a couple of things:

* `pool_label` is one of our CPG-local additions. (Which is something you “just have to know” or can figure it out via the next item)

* A useful trick for resolving these conflicts is to look at what the difference unique to our CPG branch is, i.e., the diff compared to the merge-base, _for this particular file_. So in this case:

```
$ git merge-base 8f6797b03 b7bde56d5
da6668bfd58fe915c54f052844db18975ec7abc1
$ git diff da6668bfd58fe915c54f052844db18975ec7abc1..origin/main -- batch/batch/front_end/validate.py
diff --git a/batch/batch/front_end/validate.py b/batch/batch/front_end/validate.py
index d18dd839f..52bcc405c 100644
--- a/batch/batch/front_end/validate.py
+++ b/batch/batch/front_end/validate.py
@@ -91,6 +91,7 @@ job_validator = keyed(
                 'cpu': regex(CPU_REGEXPAT, CPU_REGEX),
                 'storage': regex(STORAGE_REGEXPAT, STORAGE_REGEX),
                 'machine_type': str_type,
+                'pool_label': str_type,
                 'preemptible': bool_type,
             }
         ),
```

So adding `pool_label` to this list was our local change, and we need to re-add it in the merged version.

That was the correct handling for the `"jvm" part: turns out that was already there, it just got reformatted.